src/adapters/bidderFactory.js

```diff
@@ -250,6 +255,17 @@ export function newBidder(spec) {
     }
   });
 
+  function isUnknownBidder(responseBidder, requestBidder, adapterCode) {
+    let currentBidderSettings = !!bidderSettings.get(requestBidder, 'allowUnknownBidderCodes', adapterCode || null);
+    let unknownBiddersList = bidderSettings.get(requestBidder, 'allowedUnknownBidderCodes', adapterCode || null);
+    if (!!responseBidder && !adapterManager.aliasRegistry[responseBidder] && !!requestBidder && requestBidder !== responseBidder) {
```

Comment:
I read this block as:

if both `responseBidder` and `requestBidder` are defined and different from each other, and `responseBidder` is not an alias, then check that `responseBidder` is allowed by `bidderSettings`.

But this would allow for example the `pubmatic` adapter to set any alias as the `responseBidder`, like `appnexusAst` (an alias of `appnexus`). 

I think the logic should be: if `responseBidder` is defined and different from `requestBidder`, check that its adapter code is the same as the adapter code of `requestBidder`, or that it is allowed by `bidderSettings`.

Reply:
Hi @dgirardi ,
Lets consider 2 scenarios
1. Without alias
2. With alias

***1. Without alias***
```
pbjs.bidderSettings = {
     pubmatic: {
        allowUnknownBidderCodes: true,
        allowedUnknownBidderCodes: ['unexpected']
     }
}
pbjs.requestBids({
   adUnits: [{
       code: 'div-1'
       bids: [{
             bidder: 'pubmatic'    // the pubmatic adapter decides to set `bidResponse.bidderCode = 'unexpected'`
        }]
   }]
})
```
In above case `pubmatic` is bidder which will respond with bidderCode as `unexpected `
So input to `isUnknownBidder` will be `responseBidder as unexpected `, `requestBidder as pubmatic` and `adapterCode as pubmatic`
Here, the function will find entry for `pubmatic` and will get `unexpected ` in the `allowedUnknownBidderCodes` array so it will return false and continue with the bid.


***2. With alias***
```
pbjs.aliasBidder('pubmatic', 'foobar');
pbjs.bidderSettings = {
     pubmatic: {
        allowUnknownBidderCodes: true,
        allowedUnknownBidderCodes: ['unexpected']
     }
}
pbjs.requestBids({
   adUnits: [{
       code: 'div-1'
       bids: [{
             bidder: 'foobar'    // the pubmatic adapter decides to set `bidResponse.bidderCode = 'unexpected'`
        }]
   }]
})
```
In above case `foobar` is alias bidder of `pubmatic` adapter which will respond with bidderCode as `unexpected `
So input to `isUnknownBidder` function will be `responseBidder as unexpected `, `requestBidder as foobar` and `adapterCode as pubmatic`
Here, the function will first search for `bidderSettings` of foobar but couldn't find any so it searches for `bidderSettings` of its adapter i.e `pubmatic`. Here the entry is found and `unexpected ` is present in the `allowedUnknownBidderCodes` array so it will return false and continue with the bid.

To search for `bidderSettings` of `adapterCode` if `bidderSettings` are not present for given `bidderCode`, i have added additional parameter `adapterCode`.

Reply:
HI @dgirardi ,
Got your point. Made the related changes and also added unit test cases for it too. Please review.